source/common/router/header_parser.cc

```diff
@@ -251,6 +251,7 @@ HeaderFormatterPtr parseInternal(const envoy::config::core::v3::HeaderValue& hea
 
 HeaderParserPtr
 HeaderParser::configure(const Protobuf::RepeatedPtrField<HeaderValueOption>& headers_to_add) {
+  std::set<absl::string_view> headers_to_overwrite;
```

Comment:
Can we sink this variable into a scope so it is freed before line 358, and also rename one of the  two different headers_to_overwrite variables? It's unambiguous to the compiler but confusing to humans.

Also I think this can be a flat_hash_set.

Reply:
Changed the type to `absl::flat_hash_set`. Thanks!

Hmm, it is true that I used `headers_to_overwrite` in two places, but they are in separate functions. The scope of the first `headers_to_overwrite` is limited to `HeaderParser::configure` method and the second one is in `HeaderParser::evaluateHeaders` method. I can still change the name, but have problems with finding a good name :-)